src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -140,6 +140,33 @@ exception StandardOperatorRedefinitionWarning of string * range
 
 exception InvalidInternalsVisibleToAssemblyName of badName: string * fileName: string option
 
+//----------------------------------------------------------------------------------------------
+// Helpers for determining if/what specifiers a string has.
+// Used to decide if interpolated string can be lowered to a concat call.
+// We don't care about single- vs multi-$ strings here, because lexer took care of that already.
+//----------------------------------------------------------------------------------------------
+module private ParseString =
+    open System.Text.RegularExpressions
+
+    // Regex for strings ending with unescaped '%s'
+    // For efficiency sake using lookbehind, starting from end of the string
+    let stringSpecifier = Regex(@"$(?<=(^|[^%])(%%)*%s)", RegexOptions.Compiled)
+    let EndsWithStringSpecifierNoFlags = stringSpecifier.IsMatch
+
+    // Regex pattern for something like: %[flags][width][.precision][type]
```

Comment:
This is a nice RegEx breakdown - still more examples always help in this cases, so if you put a few examples here in the comments (e.g. with and without optional parts), that would be awesome.